clients/java/signalr/src/main/java/JsonHubProtocol.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.util.ArrayList;
+
+public class JsonHubProtocol implements IHubProtocol{
+    private JsonParser jsonParser = new JsonParser();
+    private static final String RECORD_SEPARATOR = "\u001e";
+
+    @Override
+    public String name() {
+        return "json";
+    }
+
+    @Override
+    public int version() {
+        return 1;
+    }
+
+    @Override
+    public TransferFormat transferFormat() {
+        return TransferFormat.Text;
+    }
+
+    @Override
+    public InvocationMessage[] parseMessages(String payload) {
+
+        String[] messages = payload.split(RECORD_SEPARATOR);
+        ArrayList<InvocationMessage> invocationMessages = new ArrayList<>();
+        // Empty handshake response "{}". We can ignore it
+        for (String splitMessage: messages) {
+            if (splitMessage.length() == 2) {
```

Comment:
nit: you could also `!"{}".equals(message)` for readability

Reply:
Yeah, that's better. More exact and readable

Reply:
You might want to stub that out by adding a HandshakeProtocol next and putting that outside of  the protocol implementation itself.